app/src/androidTest/java/com/github/multimatum_team/multimatum/MainSettingsActivityTest.kt

```diff
@@ -74,36 +87,51 @@ class MainSettingsActivityTest {
         initNotifEnabled: Boolean, initDarkModeEnabled: Boolean,
         clickedButtonId: Int,
         expectedFinalNotifEnabled: Boolean, expectedFinalDarkModeEnabled: Boolean
-    ) {
+    ){
+        `when`(mockSharedPreferences.getBoolean(eq(MainSettingsActivity.NOTIF_ENABLED_PREF_KEY), any()))
+            .thenReturn(initNotifEnabled)
+        `when`(mockSharedPreferences.getBoolean(eq(MainSettingsActivity.DARK_MODE_PREF_KEY), any()))
+            .thenReturn(initDarkModeEnabled)
+        val mockEditor: SharedPreferences.Editor = mock()
+        `when`(mockEditor.putBoolean(eq(MainSettingsActivity.NOTIF_ENABLED_PREF_KEY), any())).then {
+            assertEquals(expectedFinalNotifEnabled, it.getArgument(1))
+            mockEditor
+        }
+        `when`(mockEditor.putBoolean(eq(MainSettingsActivity.DARK_MODE_PREF_KEY), any())).then {
+            assertEquals(expectedFinalDarkModeEnabled, it.getArgument(1))
+            mockEditor
+        }
+        `when`(mockEditor.apply()).then { /* do nothing */ }
```

Comment:
`.then` is used to define the behavior that the mocked object should have when a call as specified in `when` occurs. Here it might work if I remove the `then` (not sure), but I think it is clearer as it is now anyway, because the expected behavior on a call is indeed to "do nothing".